examples/numalogic-simple-pipeline/src/utils.py

```diff
@@ -16,36 +15,44 @@
 LOGGER = logging.getLogger(__name__)
 
 
-@dataclass_json
 @dataclass(slots=True)
 class Payload:
     """Payload to be used for inter-vertex data transfer."""
 
-    ts_data: ArrayLike = None
-    anomaly_score: float = 0.0
-    uuid: str = None
+    uuid: str
+    arr: list[float]
+    anomaly_score: float = None
     is_artifact_valid: bool = True
 
+    def get_array(self) -> npt.NDArray[float]:
+        return np.asarray(self.arr)
 
-def save_artifact(artifact, skeys: Sequence[str], dkeys: Sequence[str]) -> None:
-    """Saves the artifact into mlflow registry and updates version."""
-    if isinstance(artifact, BaseAE):
-        ml_registry = MLflowRegistry(tracking_uri=TRACKING_URI, artifact_type="pytorch")
-    else:
-        ml_registry = MLflowRegistry(tracking_uri=TRACKING_URI, artifact_type="sklearn")
-    ml_registry.save(skeys=skeys, dkeys=dkeys, artifact=artifact)
-
-
-def load_artifact(skeys: Sequence[str], dkeys: Sequence[str], type_: str = None) -> ArtifactDict:
-    """Loads the desired artifact from mlflow registry and returns it."""
-    try:
-        if type_ == "pytorch":
-            ml_registry = MLflowRegistry(tracking_uri=TRACKING_URI, artifact_type="pytorch")
-        else:
-            ml_registry = MLflowRegistry(tracking_uri=TRACKING_URI, artifact_type="sklearn")
-        artifact_dict = ml_registry.load(skeys=skeys, dkeys=dkeys)
-    except Exception as ex:
-        LOGGER.exception("Error while loading artifact from MLFlow database: %s", ex)
-        return None
-    else:
-        return artifact_dict
+    def set_array(self, arr: list[float]) -> None:
+        self.arr = arr
+
+    def to_json(self) -> bytes:
+        """Converts the payload to json."""
+        return json.dumps(asdict(self)).encode("utf-8")
+
+    @classmethod
+    def from_json(cls, json_payload: bytes) -> Self:
+        """Converts the json payload to Payload object."""
+        return cls(**json.loads(json_payload.decode("utf-8")))
+
+
+class NumalogicUDF:
```

Comment:
We can move this to numalogic, if we all agree on the structure.

Reply:
this a good one